synapse/server_notices/resource_limits_server_notices.py

```diff
@@ -91,47 +91,77 @@ def maybe_send_server_notice_to_user(self, user_id):
-                is_auth_blocking = True
-                event_content = e.msg
-                event_limit_type = e.limit_type
+            if not self._config.mau_limit_alerting:
+                # Alerting disabled, reset room if necessary and return
+                if currently_blocked:
+                    self._remove_limit_block_notification(ref_events, user_id)
+                return
+
+            is_auth_blocking = False
+            if self._config.mau_limit_alerting:
+                try:
+                    # Normally should always pass in user_id to check_auth_blocking
+                    # if you have it, but in this case are checking what would happen
+                    # to other users if they were to arrive.
+                    yield self._auth.check_auth_blocking()
+                except ResourceLimitError as e:
+                    is_auth_blocking = True
+                    event_body = e.msg
+                    event_limit_type = e.limit_type
 
             if currently_blocked and not is_auth_blocking:
                 # Room is notifying of a block, when it ought not to be.
-                # Remove block notification
-                content = {"pinned": ref_events}
-                yield self._server_notices_manager.send_notice(
-                    user_id, content, EventTypes.Pinned, ""
-                )
-
+                yield self._remove_limit_block_notification(user_id, ref_events)
             elif not currently_blocked and is_auth_blocking:
                 # Room is not notifying of a block, when it ought to be.
-                # Add block notification
-                content = {
-                    "body": event_content,
-                    "msgtype": ServerNoticeMsgType,
-                    "server_notice_type": ServerNoticeLimitReached,
-                    "admin_contact": self._config.admin_contact,
-                    "limit_type": event_limit_type,
-                }
-                event = yield self._server_notices_manager.send_notice(
-                    user_id, content, EventTypes.Message
-                )
-
-                content = {"pinned": [event.event_id]}
-                yield self._server_notices_manager.send_notice(
-                    user_id, content, EventTypes.Pinned, ""
+                yield self._apply_limit_block_notification(
+                    user_id, event_body, event_limit_type
                 )
-
         except SynapseError as e:
             logger.error("Error sending resource limits server notice: %s", e)
 
+    @defer.inlineCallbacks
+    def _remove_limit_block_notification(self, user_id, ref_events):
+        """Utility method to remove limit block notifications from the server
+        notices room.
+
+        Args:
+            user_id (str): user to notify
+            ref_events ([]): The list of pinned events that are unrelated to
+            limit blocking and need to be preserved.
+        """
+        content = {"pinned": ref_events}
```

Comment:
this seems to be an existing problem (so not necessarily for fixing as part of this PR), but why do we preserve the other events in this case and not in the `_apply_limit_block_notification` case?

Reply:
From memory the idea was to shut everything down and so all tags should be discarded, but in retrospect this seems unhelpful.

Reply:
fixed

Reply:
unfixed, will address in a separate PR.